### PR TITLE
minor dev guide tweaks

### LINKED ISSUE #4076: In Developer Guide, it's not obvious that "following pages" are indexed in the sidebar on the left
In http://bokeh.pydata.org/en/latest/docs/dev_guide.html#bokehjs there is this text:

    Hopefully, you will find all the information you need in the following pages. 

However, there is no additional pages after the text, so it just looks like the docs are blank.  Rather, the sidebar on the left contains the index, and it's not obvious that this is where people should look.  So, either we should rephrase the text to say "the sidebar on the left", or somehow we duplicate the index in the text body itself.

### PR DESCRIPTION
**Pull request by bryevdv**
- [x] issues: fixes #4076
- [x] tests added / passed
- [x] release document entry (if new feature or API change)

In this PR:

* make it clear the dev guide has content
* fix some broken references
* rename section "objects" -> "models" for consistency
* delete hopelessly out of date BokehJS info

### CODE REVIEW

**Comment by bryevdv** [comment]
Screenshot of the new dev guide landing page:

screen shot 2016-04-30 at 12 46 51 am

**Comment by bryevdv** [comment]
ping @birdsarah @mattpap @bokeh/dev ready for review

File: sphinx/source/docs/dev_guide/bokehjs.rst [hunk 1]
```diff
@@ -43,113 +43,14 @@ web workers, asm.js, SIMD, and parallel JS (e.g. River Trail).
...
-.. raw:: html
-
-    <iframe
-        width="100%"
-        height="700"
-        src="http://jsfiddle.net/bokeh/K8P4P/embedded/result,js/"
-        allowfullscreen="allowfullscreen"
-        frameborder="0"
-    ></iframe>
-
```
Comment (bryevdv) [inline sphinx/source/docs/dev_guide/bokehjs.rst]: Intend to add more information back here as part of separate PR to document new BokehJS interface.

**Comment by birdsarah** [comment]
I haven't run it, but it looks good to me. Especially the front page.

Thanks for doing this!

**Pull request merged**